core/pkg/workers/http/http.go

```diff
@@ -38,14 +38,35 @@ type HttpRequestHandler interface {
 
 // IsPortOpen returns true if the port is open, false otherwise
 // returns false if the timeout elapses before a connection is established
-func IsPortOpen(host string, timeout time.Duration) bool {
-	conn, err := net.DialTimeout("tcp", host, timeout)
-	if err != nil {
-		return false
-	}
+func IsPortOpenWithRetry(host string, connectionTimeout time.Duration, maxRetries int) error {
+	// Create a channel for signaling when the connection is successful or max retries are reached
+	done := make(chan error, 1)
+	defer close(done)
+
+	// Create a ticker to periodically attempt the connection
+	ticker := time.NewTicker(time.Second) // Adjust the interval as needed
```

Comment:
do we want this to be a bit tighter on timing?

Reply:
reduce to 50ms